### PR TITLE
Enable all net48 integration tests and stabilise tests

### PR DESCRIPTION
**Pull request by scott-xu**

### CODE REVIEW

**Comment by Rob-Hague** [comment]
We can merge this one next assuming it passes. No need to create and resolve conflicts in the other branches

**Comment by scott-xu** [comment]
Deal 🥂

**Comment by scott-xu** [comment]
`public static System.Security.Cryptography.ECDsa Create(System.Security.Cryptography.ECParameters parameters)` applies to net47 and above.

See https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.ecdsa.create?view=net-8.0#system-security-cryptography-ecdsa-create(system-security-cryptography-ecparameters)

**Comment by scott-xu** [comment]
Unfortunately, Mono doesn't implement `ECDsa`. See https://github.com/mono/mono/blob/main/mcs/class/referencesource/System.Core/System/Security/Cryptography/ECDsa.cs#L32

**Comment by WojciechNagorski** [comment]
So we can run them on Windows. What do you think?

**Comment by Rob-Hague** [comment]
> Unfortunately, Mono doesn't implement `ECDsa`. See https://github.com/mono/mono/blob/main/mcs/class/referencesource/System.Core/System/Security/Cryptography/ECDsa.cs#L32

It is the reason behind #1406 

> So we can run them on Windows. What do you think?

I tried a while back. The problem is running a Linux container on a Windows host, which doesn't seem so easy on CI machines (the way to do it is WSL2 AFAIK but most Windows server images do not come with WSL2). But I'd welcome anyone to try because I'd very much like to see it happen.

**Comment by Rob-Hague** [comment]
It might be possible now that Windows Server 2022 includes WSL2. The problems I ran into were seemingly related to LCOW (https://github.com/sshnet/SSH.NET/pull/1220#issuecomment-1773930792). I wouldn't know if it is possible to run docker via WSL2 rather than LCOW in appveyor (I don't really know much about this stuff)

**Comment by Rob-Hague** [comment]
I think there's some work involved here to make sure we don't introduce more unstable tests

**Comment by scott-xu** [comment]
> > Unfortunately, Mono doesn't implement `ECDsa`. See https://github.com/mono/mono/blob/main/mcs/class/referencesource/System.Core/System/Security/Cryptography/ECDsa.cs#L32
> 
> It is the reason behind #1406
> 

We can consider using BouncyCastle as a fallback when in Mono. (https://www.mono-project.com/docs/faq/technical/#how-can-i-detect-if-am-running-in-mono)

Then we can split the code into 3 implementations.
1. Bcl implementation (.NET Framework 4.7+ and not Mono)
2. Cng implementation (.NET Framework 4.6.2 and not Mono)
3. BouncyCastle implementation (.NET Framework and Mono)

I can have a try when I have time.

**Comment by scott-xu** [comment]
It seems not easy to split `EcdsaKey` into three implementations because it is public already with `ECDsaCng Ecdsa` and `CngAlgorithm HashAlgorithm` properties when target to .NET Framework.

**Comment by Rob-Hague** [comment]
I kind of expected mono to use a .NET Standard build, but I guess it can use a .NET Framework build as well. The error in #1406 comes from calling `ECDsa.Create(ECParameters)` which only occurs #if !NET_FRAMEWORK.

Anyway I think it's fine to tweak the public members at this level if needed, especially if it doesn't work in the first place.

I'll run the CI on this a couple more times to check for any more misbehaving tests

File: test/Renci.SshNet.IntegrationTests/SshTests.cs [hunk 1]
```diff
@@ -655,17 +655,32 @@ public void Ssh_RemotePortForwarding()
             var endpoint1 = new IPEndPoint(ipv4HostAddress, 10000);
             var endpoint2 = new IPEndPoint(ipv4HostAddress, 10001);
 
+            var isBytesReceivedOnLister1 = false;
+            var isBytesReceivedOnLister2 = false;
```
Comment (Rob-Hague) [inline test/Renci.SshNet.IntegrationTests/SshTests.cs]:
```suggestion
            var isBytesReceivedOnListener1 = false;
            var isBytesReceivedOnListener2 = false;
```

**Comment by scott-xu** [comment]
Any idea to stabilize this test? @Rob-Hague 
```
  Failed Test_ExecuteAsync_CancellationToken [1 s]
  Error Message:
   Assert.AreEqual failed. Expected:<TERM>. Actual:<(null)>. 
  Stack Trace:
     at Renci.SshNet.IntegrationTests.OldIntegrationTests.SshCommandTest.Test_ExecuteAsync_CancellationToken() in /home/appveyor/projects/ssh-net/test/Renci.SshNet.IntegrationTests/OldIntegrationTests/SshCommandTest.cs:line 139
```
https://ci.appveyor.com/project/drieseng/ssh-net/builds/50322490/job/yo6pooiew5j86wcw

The 500ms wait may not enough in unit test.
See https://github.com/sshnet/SSH.NET/blob/develop/src/Renci.SshNet/SshCommand.cs#L446

**Comment by Rob-Hague** [comment]
Hmm, I would not expect the 500ms to be relevant here because we know OpenSSH implements signals. I think we can ignore that one and see if it happens more often

**Review by Rob-Hague — APPROVED** [review]
Thanks

**Pull request merged**